ucsm_apis/admin/callhome.py

```diff
@@ -0,0 +1,803 @@
+    """
+    try:
+        mo = call_home_profile_get(handle, name,
+                                   caller="call_home_profile_exist")
+    except UcsOperationError:
+        return (False, None)
+    mo_exists = mo.check_prop_match(**kwargs)
+    return (mo_exists, mo if mo_exists else None)
+
+
+def call_home_profile_modify(handle, name, **kwargs):
+    """
+    Modifies callhome profile.
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of callhome profile
+        **kwargs: key-value pair of managed object(MO) property and value, Use
+                  'print(ucsccoreutils.get_meta_info(<classid>).config_props)'
+                  to get all configurable properties of class
+
+    Returns:
+        CallhomeProfile : ManagedObject
+
+    Raises:
+        UcsOperationError: If CallhomeProfile is not present
+
+    Example:
+        call_home_profile_modify(handle, name="callhomeprofile", format="xml")
+    """
+    mo = call_home_profile_get(handle, name, caller="call_home_profile_modify")
+    mo.set_prop_multiple(**kwargs)
+    handle.set_mo(mo)
+    handle.commit()
+    return mo
+
+
+def call_home_profile_delete(handle, name):
+    """
+    Deletes callhome profile.
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of callhome profile
+        **kwargs: key-value pair of managed object(MO) property and value, Use
+                  'print(ucsccoreutils.get_meta_info(<classid>).config_props)'
+                  to get all configurable properties of class
+
+    Returns:
+        None
+
+    Raises:
+        UcsOperationError: If CallhomeProfile is not present
+
+    Example:
+        call_home_profile_delete(handle, name="callhomeprofile")
+    """
+    mo = call_home_profile_get(handle, name, caller="call_home_profile_delete")
+    handle.remove_mo(mo)
+    handle.commit()
+
+
+def call_home_profile_email_add(handle, profile_name, email, **kwargs):
+    """
+    Adds email to callhome profile.
+
+    Args:
+        handle (UcsHandle)
+        profile_name (string): name of callhome profile
+        email (string): receipient email address
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+
+    Returns:
+        CallhomeDest : ManagedObject
+
+    Raises:
+        UcsOperationError: If CallhomeProfile is not present
+
+    Example:
+        call_home_profile_email_add(handle, profile_name="callhomeprofile",
+                                    email="ciscoucs@cisco.com")
+    """
+    from ucsmsdk.mometa.callhome.CallhomeDest import CallhomeDest
+
+    profile = call_home_profile_get(handle, profile_name,
+                                    caller="call_home_profile_email_add")
+    mo = CallhomeDest(parent_mo_or_dn=profile, email=email)
+    mo.set_prop_multiple(**kwargs)
+    handle.add_mo(mo, True)
```

Comment:
we should try to use namedParam going forward. This will ensure that the code continues to work even if we refactor order of args in `add_mo`